File: DataServer/GrpcService/converters/PlayerConverter.cs
using DTO;
using Google.Protobuf.Collections;
using grpcProtoFiles;

namespace GrpcService.converters;

public class PlayerConverter
{
    public PlayerDTO ToDTO(Player player) {
        return new PlayerDTO {
            Name = player.Name
        };
    }
    
    public List<PlayerDTO> ToDTOList(List<Player> players)
    {
        List<PlayerDTO> _players = new List<PlayerDTO>();
        foreach (var player in players)
        {
            _players.Add(ToDTO(player));
        }

        return _players;
    }

    public Player ToProto(PlayerDTO player) {
        return new Player {
            Name = player.Name
        };
    }
    
    public RepeatedField<Player> ToProtoList(List<PlayerDTO> players)
    {
        List<Player> _players = new List<Player>();
        foreach (var player in players)
        {
            _players.Add(ToProto(player));
        }

        return new RepeatedField<Player>(){ _players };
    }
}

File: Server/src/main/java/com/example/server/grpc/converters/PlayerConverter.java
package com.example.server.grpc.converters;

import com.example.server.dto.PlayerDTO;
import grpcProtoFiles.PlayerOuterClass;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class PlayerConverter {
    public PlayerDTO toDto(PlayerOuterClass.Player player) {
        return PlayerDTO.builder()
                .name(player.getName())
                .build();
    }

    public List<PlayerDTO> toDtoList(List<PlayerOuterClass.Player> players) {
        return players.stream().map(player -> toDto(player)).toList();
    }

    public PlayerOuterClass.Player toProto(PlayerDTO player) {
        return PlayerOuterClass.Player.newBuilder()
                .setName(player.getName())
                .build();
    }

    public List<PlayerOuterClass.Player> toProtoList(List<PlayerDTO> players) {
        return players.stream().map(player -> toProto(player)).toList();
    }
}
